tools/distpartitioning/convert_partition.py

```diff
@@ -450,56 +459,156 @@ def create_dgl_object(
     )
 
     # create the graph here now.
-    part_graph = dgl.graph(
-        data=(part_local_src_id, part_local_dst_id), num_nodes=len(uniq_ids)
-    )
-    part_graph.edata[dgl.EID] = th.arange(
-        edgeid_offset,
-        edgeid_offset + part_graph.num_edges(),
-        dtype=th.int64,
-    )
-    part_graph.edata[dgl.ETYPE] = th.as_tensor(
-        etype_ids, dtype=RESERVED_FIELD_DTYPE[dgl.ETYPE]
-    )
-    part_graph.edata["inner_edge"] = th.ones(
-        part_graph.num_edges(), dtype=RESERVED_FIELD_DTYPE["inner_edge"]
-    )
+    # create the graph here now.
+    if use_graphbolt:
+        edge_attr = {}
+        num_edges = len(part_local_dst_id)
+        edge_attr[dgl.EID] = th.arange(
```

Comment:
we should let user have control on whether each node/edge attributes to be saved during partition. This could save non-trivial memory for some use cases. But to be clear, I think it's ok to add such control(same as `partition_graph()`) in a following separate PR.